src/Form/Form.js

```diff
@@ -386,26 +387,36 @@ class Form {
         }
     }
 
+    /**
+     * The function looks for form's control elements, and returns them if they're iterable.
+     * @param {*} selector
+     */
+    getFormControlElements(selector) {
+        // Some sites seem to be overriding `form.elements`, so we need to check if it's still iterable.
+        if (this.form instanceof HTMLFormElement && this.form.elements != null && Symbol.iterator in Object(this.form.elements)) {
+            // For form elements we use .elements to catch fields outside the form itself using the form attribute.
+            // It also catches all elements when the markup is broken.
+            // We use .filter to avoid specific types of elements.
+            const formControls = [...this.form.elements].filter((el) => el.matches(selector));
+            return [...formControls];
+        } else {
+            return null;
+        }
+    }
+
     categorizeInputs() {
         const selector = this.matching.cssSelector('formInputsSelector');
         // If there's no form container and it's just a lonely input field (this.form is an input field)
         if (this.form.matches(selector)) {
             this.addInput(this.form);
         } else {
-            /** @type {Element[] | NodeList} */
-            let foundInputs = [];
-            // Some sites seem to be overriding `form.elements`, so we need to check if it's still iterable.
-            if (this.form instanceof HTMLFormElement && this.form.elements != null && Symbol.iterator in Object(this.form.elements)) {
-                // For form elements we use .elements to catch fields outside the form itself using the form attribute.
-                // It also catches all elements when the markup is broken.
-                // We use .filter to avoid fieldset, button, textarea etc.
-                const formElements = [...this.form.elements].filter((el) => el.matches(selector));
-                // If there are no form elements, we try to look for all
-                // enclosed elements within the form.
-                foundInputs = formElements.length > 0 ? formElements : findEnclosedElements(this.form, selector);
-            } else {
-                foundInputs = this.form.querySelectorAll(selector);
-            }
+            // Attempt to get form's control elements first as it can catch elements when markup is broke, or if the fields are outside the form.
+            // Other wise use queryFormElements, that can scan for shadow tree.
+            const formControlElements = this.getFormControlElements(selector);
+            const foundInputs =
+                formControlElements != null
+                    ? [...formControlElements, ...findElementsInShadowTree(this.form, selector)]
+                    : queryFormElements(this.form, selector, true);
```

Comment:
It's clear here now that we need to try to get elements first using `.elements` for a specific reason. Otherwise we do regular querying (with shadow `true`)